x-pack/plugins/fleet/server/integration_tests/ha_setup.test.ts

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Path from 'path';
+
+import { range } from 'lodash';
+
+import type { ISavedObjectsRepository } from 'src/core/server';
+import * as kbnTestServer from 'src/core/test_helpers/kbn_server';
+
+import type {
+  AgentPolicySOAttributes,
+  Installation,
+  OutputSOAttributes,
+  PackagePolicySOAttributes,
+} from '../types';
+
+const logFilePath = Path.join(__dirname, 'logs.log');
+
+type Root = ReturnType<typeof kbnTestServer.createRoot>;
+
+const startAndWaitForFleetSetup = async (root: Root) => {
+  const start = await root.start();
+
+  const isFleetSetupRunning = async () => {
+    const statusApi = kbnTestServer.getSupertest(root, 'get', '/api/status');
+    const resp = await statusApi.send();
+    const fleetStatus = resp.body?.status?.plugins?.fleet;
+    if (fleetStatus?.meta?.error) {
+      throw new Error(`Setup failed: ${JSON.stringify(fleetStatus)}`);
+    }
+
+    return !fleetStatus || fleetStatus?.summary === 'Fleet is setting up';
+  };
+
+  while (await isFleetSetupRunning()) {
+    await new Promise((resolve) => setTimeout(resolve, 2000));
+  }
+
+  return start;
+};
+
+const createAndSetupRoot = async (config?: object) => {
+  const root = kbnTestServer.createRootWithCorePlugins(
```

Comment:
We should probably use a docker registry for these tests to avoid some flakyness I created an helper to do so in https://github.com/elastic/kibana/pull/124405

Reply:
If we had the bundled packages copied in during `yarn kbn bootstrap` that would also work, agreed this is best for now.